src/Microsoft.AspNet.OData.Shared/TypelessPatchMethodHandler.cs

```diff
@@ -0,0 +1,196 @@
+﻿using Microsoft.OData.Edm;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.OData
+{    
+   
+    /// <summary>
+    /// Handler Class to handle users methods for create, delete and update
+    /// </summary>
+    public abstract class TypelessPatchMethodHandler 
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>
+        /// <param name="changedObj">Changed object which can be appied on creted object, optional</param>
+        /// <param name="createdObject">The created object (Typeless)</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryCreate(IEdmChangedObject changedObj, out EdmStructuredObject createdObject, out string errorMessage);
+
+        /// <summary>
+        /// TryGet method to which pointer to TryGet method can be assigned to.  This tries to Get based on a keyvalues.
+        /// </summary>
+        /// <param name="keyValues">Key value pair for the entity keys</param>        
+        /// <param name="originalObject">Object to return</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage);
```

Comment:
>EdmStructuredObject [](start = 86, length = 19)

For TryGet and TryCreate, is there any advantage to returning an EdmStructuredObject rather than an IEdmStructuredObject?